src/dxvk/dxvk_fence.cpp

```diff
@@ -56,80 +57,94 @@ namespace dxvk {
         Logger::warn(str::format("Importing semaphores of type ", info.sharedType, " not supported by device"));
       }
     }
-
-    m_thread = dxvk::thread([this] { run(); });
   }
 
 
   DxvkFence::~DxvkFence() {
-    m_stop.store(true);
-    m_thread.join();
+    if (m_running.exchange(false, std::memory_order_acq_rel)) {
```

Comment:
need to lock here, or otherwise we'll risk setting this and notifying the cond var before the worker thread actually waits on the cond var.

Reply:
Would that actually be a problem? The worker thread would check if m_running is set as part of the condition variable predicate and skip waiting.

Reply:
This series of events is possible:
- Thread reads m_running (true)
- This thread writes m_running.
- This thread calls notify_one().
- Thread calls wait().
- Deadlock

Reply:
If the write to m_running happens "in between" the m_running check on the worker thread and the wait call?